spec/unit/form_builder_spec.rb

```diff
@@ -277,6 +277,43 @@ def build_form(options = {}, form_object = Post.new, &block)
 
   end
 
+  context "with a has_one relation on an author's profile" do
+    let :body do
+      author = user()
+      build_form do |f|
+        f.inputs do
+          f.input :title
+          f.input :body
+        end
+        f.form_builder.instance_eval do
+          @object.author = author
```

Comment:
I needed to test against author.profile being undefined (the bug) and where author.profile was set to a new record (the suggested work around).